samples/word/50-document/insert-external-document.yaml

```
@@ -0,0 +1,101 @@
+order: 2
+id: word-document-insert-external-document
+name: Insert an external document
+description: Inserts an external document into the current document.
+host: WORD
+api_set:
+    WordApi: 1.1
+script:
+    content: |
+        $("#file").change(getBase64);
+        $("#insert-document").click(() => tryCatch(insertDocument));
+
+        let externalDocument;
+
+        async function insertDocument() {
+          // Update the text of the current document with the text from another document passed in as a Base64-encoded string.
+          await Word.run(async (context) => {
+            // Use the Base64-encoded string representation of the selected .docx file.
+            const externalDoc = context.application.createDocument(externalDocument);
+            await context.sync();
+
+            const externalDocBody = externalDoc.body;
+            externalDocBody.load("text");
+            await context.sync();
+
+            const externalDocBodyText = externalDocBody.text;
+            const currentDocBody = context.document.body;
+            currentDocBody.insertText(externalDocBodyText, Word.InsertLocation.start);
+            await context.sync();
+          });
+        }
+
+        async function getBase64() {
+          // Retrieve the file and set up an HTML FileReader element.
+          const myFile = <HTMLInputElement>document.getElementById("file");
+          const reader = new FileReader();
+
+          reader.onload = (event) => {
+            // Remove the metadata before the Base64-encoded string.
+            const startIndex = reader.result.toString().indexOf("base64,");
+            externalDocument = reader.result.toString().substr(startIndex + 7);
+          };
+
+          // Read the file as a data URL so that we can parse the Base64-encoded string.
+          reader.readAsDataURL(myFile.files[0]);
+        }
+
+        /** Default helper for invoking an action and handling errors. */
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            <p>This sample shows how to get the body text from an external document into the current document.</p>
```

"Insert" articulates the scenario a bit more.
```suggestion
            <p>This sample shows how to insert the body text from an external document into the current document.</p>
```